src/pages/workspace/card/issueNew/AssigneeStep.tsx

```diff
@@ -1,30 +1,91 @@
 import Text from '@components/Text';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
+import {formatPhoneNumber} from '@libs/LocalePhoneNumber';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
 import Navigation from '@navigation/Navigation';
 import * as Card from '@userActions/Card';
 import CONST from '@src/CONST';
-import ONYXKEYS from '@src/ONYXKEYS';
+import type * as OnyxTypes from '@src/types/onyx';
 
-function AssigneeStep() {
+type AssigneeStepProps = {
+    // The policy that the card will be issued under
+    policy: OnyxEntry<OnyxTypes.Policy>;
+};
+
+function AssigneeStep({policy}: AssigneeStepProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
+    const [searchTerm, setSearchTerm] = useState('');
 
-    const submit = () => {
-        // TODO: the logic will be created in https://github.com/Expensify/App/issues/44309
-        Card.setIssueNewCardStep(CONST.EXPENSIFY_CARD.STEP.CARD_TYPE);
+    const submit = (assignee: ListItem) => {
+        Card.setIssueNewCardDataAndGoToStep({assigneeEmail: assignee?.login ?? ''}, CONST.EXPENSIFY_CARD.STEP.CARD_TYPE);
     };
 
     const handleBackButtonPress = () => {
         Navigation.goBack();
     };
 
+    const shouldShowSearchInput = policy?.employeeList && Object.keys(policy.employeeList).length > 8;
+    const textInputLabel = shouldShowSearchInput ? translate('workspace.card.issueNewCard.findMember') : undefined;
+
+    const membersDetails = useMemo(() => {
+        if (!policy?.employeeList) {
+            return [];
+        }
+
+        return Object.keys(policy.employeeList)
+            .map(PersonalDetailsUtils.getPersonalDetailByEmail)
+            .map((detail) => ({
+                keyForList: detail?.login,
+                text: detail?.displayName,
+                alternateText: detail?.login,
+                login: detail?.login,
+                accountID: detail?.accountID,
+                icons: [
+                    {
+                        source: detail?.avatar ?? Expensicons.FallbackAvatar,
+                        name: formatPhoneNumber(detail?.login ?? ''),
+                        type: CONST.ICON_TYPE_AVATAR,
+                        id: detail?.accountID,
+                    },
+                ],
+            }));
+    }, [policy?.employeeList]);
+
+    const sections = useMemo(() => {
+        if (!searchTerm) {
+            return [
+                {
+                    data: membersDetails,
+                    shouldShow: true,
+                },
+            ];
+        }
+
+        const searchValue = OptionsListUtils.getSearchValueForPhoneOrEmail(searchTerm).toLowerCase();
+        const filteredOptions = membersDetails.filter((option) => !!option.text?.toLowerCase().includes(searchValue) || !!option.alternateText?.toLowerCase().includes(searchValue));
```

Comment:
@koko57  Could you consider to use OptionsListUtils.filterOptions to filter members for consistency? And also use debounce for searchValue

Reply:
@DylanDylann I don't think we can use filterOptions here - we don't use it in WorkspaceMembersPage either and we don't have the Options object like in WorkspaceInvitePage or MoneyRequestParticipantsSelector